### PR TITLE
New board support process rcxvup

### PR DESCRIPTION
**Pull request by acastellane**
After RCXVUP bring up we have a stable code.
Missing :
* User/factory mechanism (reset, user address and switching to be fixed)
* second Bank & QDR are not available

Signed-off-by: Alexandre Castellane alexandre.castellane@fr.ibm.com

### CODE REVIEW

File: hardware/doc/SNAP-Registers.md [hunk 1]
```diff
@@ -193,6 +193,7 @@ Address: 0x0000030
   15..9  RO: Reserved
       8  RO: NVMe enabled
    7..0  RO: Card type:
+             0x11 : RCXVUP
```
Comment (boekholt) [inline hardware/doc/SNAP-Registers.md]: The corresponding change is missing in the slave context register section (lines 505 ff.)

File: hardware/setup/create_framework.tcl [hunk 2]
```diff
@@ -283,6 +283,11 @@ if { ($fpga_card == "N250SP") && ($capi_bsp_dir != "not defined") } {
   add_files -fileset constrs_1 -norecurse $root_dir/setup/$fpga_card/snap_$fpga_card.xdc
 #  add_files -fileset constrs_1 -norecurse $root_dir/setup/$fpga_card/capi_bsp_pblock.xdc
 }
+if { ($fpga_card == "RCXVUP") && ($capi_bsp_dir != "not defined") } {
```
Comment (boekholt) [inline hardware/setup/create_framework.tcl]: Could be added to the statement above (the RCXVUP specifics are defined via environment variable `$fpga_card`)

File: hardware/setup/create_framework.tcl [hunk 3]
```diff
@@ -304,6 +309,11 @@ if { $fpga_card == "ADKU3" } {
     add_files -fileset constrs_1 -norecurse $root_dir/setup/AD8K5/snap_ddr4_b0pins.xdc
     set_property used_in_synthesis false [get_files $root_dir/setup/AD8K5/snap_ddr4_b0pins.xdc]
   }
+} elseif { ($fpga_card == "RCXVUP") } {
```
Comment (boekholt) [inline hardware/setup/create_framework.tcl]: Could be combined with the statement below...

File: hardware/snap_check_psl [hunk 4]
```diff
@@ -25,6 +25,8 @@ if [ $# -gt 1 ]; then
 
   if [ "$FPGACARD" == "N250SP" ]; then
     export PSL_DCP_TYPE=$FPGACARD
+  elif [ "$FPGACARD" == "RCXVUP" ]; then
```
Comment (boekholt) [inline hardware/snap_check_psl]: combine with statement above?

File: snap_config [hunk 5]
```diff
@@ -53,6 +53,9 @@ while [ -z "$SETUP_DONE" ]; do
     "AD8K5" )
       SNAP_CFG="CONFIG_AD8K5=y\n""CONFIG_FPGACHIP=xcku115-flva1517-2-e\n"
       ;;
+    "RCXVUP" )
```
Comment (boekholt) [inline snap_config]: The file snap_config is not used anywhere and got removed

File: defconfig/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+Option 1 (default): use the Kconfig menu by typing: make snap_config
+ This will allow you to define a specific configuration which will be saved in the .snap_config file
+ All commands make model/sim/image will use this .snap_config file
+
+Option 2: do not use the Kconfig menu but use a specific saved configuration file (automated test -jenkins)
+ Build a .snap_config file using the Kconfig menu  (see option 1)
+ Then move and rename the .snap_config file to CARD.action.defconfig into the snap/defconfig directory
+ All commands will then be: 
+   make CARD.action.defconfig to set the .snap_config with the CARD.action.defconfig configuration
+   menu will wait for an enter key or a choice
```
[the remark this answers is not in the document]
Comment (bmesnet) [reply defconfig/README.md]: @sven I may use it a wrong way or with not all arguments but that's the only way I have been able to have it worked

File: hardware/doc/SNAP-Registers.md (hunk 1 above)
Comment (bmesnet) [reply hardware/doc/SNAP-Registers.md]: corrected.

File: hardware/setup/create_framework.tcl (hunk 3 above)
Comment (bmesnet) [reply hardware/setup/create_framework.tcl]: combined

File: hardware/setup/create_framework.tcl (hunk 2 above)
Comment (bmesnet) [reply hardware/setup/create_framework.tcl]: combined

File: snap_config (hunk 5 above)
Comment (bmesnet) [reply snap_config]: removed

File: defconfig/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (bmesnet) [reply defconfig/README.md]: correction done in README.md

File: hardware/setup/create_framework.tcl [hunk 7]
```diff
@@ -279,8 +279,8 @@ if { ($fpga_card == "N250SP") && ($capi_bsp_dir != "not defined") } {
 puts "                        importing XDCs"
 
 # Board Support XDC
-if { ($fpga_card == "N250SP") && ($capi_bsp_dir != "not defined") } {
-  puts "                        importing N250SP Board support XDCs"
+if { (($fpga_card == "N250SP") || ($fpga_card == "RCXVUP")) && ($capi_bsp_dir != "not defined") } {
+  puts "                      importing specific Board support XDCs"
```
Comment (boekholt) [inline hardware/setup/create_framework.tcl]: The format of the message is changed compared to master (removing the two spaces in front of `importing` - maybe re-introducing a previous formatting error).
Thomas won't like that the message is no longer aligned to the other messages in this level ... ;-)

File: defconfig/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (bmesnet) [reply defconfig/README.md]: lost during conflict solve. Corrected.

File: hardware/setup/create_framework.tcl (hunk 7 above)
Comment (bmesnet) [reply hardware/setup/create_framework.tcl]: good catch . corrected

File: doc/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# SNAP Documentation
+
+# 1. Overview
```
[the remark this answers is not in the document]
Comment (bmesnet) [reply doc/README.md]: corrected

File: README.md [hunk 9]
```diff
@@ -47,6 +47,7 @@ As of now, the following FPGA cards can be used with SNAP:
 * Alpha-Data ADM-PCIE-8K5 (rev2) http://www.alpha-data.com/dcp/products.php?product=adm-pcie-8k5
 * Nallatech 250S-2T with two on-card NVMe M.2 960GB drives http://www.nallatech.com/250s
 * Semptian NSA121B http://www.semptian.com/proinfo/10.html
+* ReflexCES XpressVUP-LP9PT https://www.reflexces.com/products-solutions/other-cots-boards/xilinx/xpressvup
```
[the remark this answers is not in the document]
Comment (bmesnet) [reply README.md]: changed Kconfig to reflect experimental mode of RCXVUP as N250SP card

**Comment by bmesnet** [comment]
@boekholt . This should be ok, if i didn't missed anything this time. Thx

**Pull request merged**